src/libmw/src/file/File.cpp

```diff
@@ -0,0 +1,205 @@
+#include <mw/file/File.h>
+#include <mw/common/Logger.h>
+
+#if defined(_WIN32)
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+void File::Create()
+{
+    m_path.GetParent().CreateDir();
+
+    std::ifstream inFile(m_path.m_path, std::ios::in | std::ifstream::ate | std::ifstream::binary);
+    if (inFile.is_open()) {
+        inFile.close();
+    } else {
+        LOG_INFO_F("File {} does not exist. Creating it now.", m_path);
+        std::ofstream outFile(m_path.m_path, std::ios::out | std::ios::binary | std::ios::trunc);
+        if (!outFile.is_open()) {
+            ThrowFile_F("Failed to create file: {}", m_path);
+        }
+        outFile.close();
+    }
+}
+
+bool File::Exists() const
+{
+    return m_path.Exists() && !m_path.IsDirectory();
+}
+
+void File::Truncate(const uint64_t size)
+{
+    bool success = false;
+
+#if defined(WIN32)
+    HANDLE hFile = CreateFile(
+        m_path.ToString().c_str(),
+        GENERIC_WRITE,
+        FILE_SHARE_READ,
+        NULL,
+        OPEN_EXISTING,
+        FILE_ATTRIBUTE_NORMAL,
+        NULL
+    );
+
+    LARGE_INTEGER li;
+    li.QuadPart = size;
+    success = SetFilePointerEx(hFile, li, NULL, FILE_BEGIN) && SetEndOfFile(hFile);
+
+    CloseHandle(hFile);
+#else
+    success = (truncate(m_path.ToString().c_str(), size) == 0);
+#endif
+
+    if (!success) {
+        ThrowFile_F("Failed to truncate {}", m_path);
+    }
+}
+
+void File::Rename(const std::string& filename)
+{
+    if (m_path.IsDirectory()) {
+        // FUTURE: Support renaming directories?
+        ThrowFile("Rename not implemented for directories");
+    }
+
+    const FilePath parent(m_path.GetParent());
+    if (parent == m_path) {
+        ThrowFile_F("Can't find parent path for {}", *this);
+    }
+
+    const FilePath destination = parent.GetChild(filename);
+    if (destination.Exists()) {
+        destination.Remove();
+    }
+
+    std::error_code ec;
+    filesystem::rename(m_path.m_path, destination.m_path, ec);
+    if (ec) {
+        ThrowFile_F("Failed to rename {} to {}", *this, destination);
+    }
+
+    m_path = destination.m_path;
+}
+
+std::vector<uint8_t> File::ReadBytes() const
```

Comment:
Should be implemented using File::ReadBytes(const size_t startIndex, const size_t numBytes)